### DISCUSSION TITLE
Rendering of Subcontracting  (BT-730 BT-553 BT-731 BT-555)

### DISCUSSION THREAD

**Discussion by Wout3P** [OP-TED/eForms-SDK#383] [Q&A] [answered]
Hello all,

I'm generating a preview for a notice 29 in sdk1.5 via the render notice API.
In the xml I specify following information about subcontracting :
`<efac:SubcontractingTerm>
<efbc:TermAmount currencyID="EUR">50000</efbc:TermAmount>
<efbc:TermCode listName="applicability">yes</efbc:TermCode>
<efbc:PercentageKnownIndicator>false</efbc:PercentageKnownIndicator>
<efbc:ValueKnownIndicator>true</efbc:ValueKnownIndicator>
</efac:SubcontractingTerm>`

I specify an amount and the value known indicator.
The according BT'are wel generated: BT-730 / BT-553.

However the labels for the percentage (BT-731 / BT-555) are also generated.
I would expect the labels to hidden since there is no information about percentages.

Is this a rendering problem? Will this get fixed? :)

Thanks for helping,

**Comment by rouschr** [comment]
Maybe @pdonohoe knows about this?

**Reply by pdonohoe** [reply, marked as answer]
Dear @Wout3P,

I can't comment on the way these fields were rendered at the time you submitted the XML, I don't have direct access to the rendering code used at that time.

However using the XML code you mention above, today the label for BT-555 would not be shown, as the corresponding element was not included in the XML. Only the label for BT-731 would be shown. Viewed in English, it would display:
Subcontracting percentage is known: No

This is because you included the element for BT-731 in your XML:
<efbc:PercentageKnownIndicator>false</efbc:PercentageKnownIndicator>

BT-731 is optional. If the element <efbc:PercentageKnownIndicator> is not included in the XML, then the line would not be displayed.
